azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -421,9 +426,19 @@ def _log_error_if_none(attr_name, value):
         if ext_handler.properties.state in (None, ""):
             raise ExtensionConfigError("Received empty Extensions.Plugins.Plugin.state, failing Handler")
 
-        location = getattrib(plugin, "location")
-        failover_location = getattrib(plugin, "failoverlocation")
-        for uri in [location, failover_location]:
+        def getattrib_wrapped_in_list(node, attr_name):
+            attr = getattrib(node, attr_name)
+            return [attr] if attr else []
+
+        location = getattrib_wrapped_in_list(plugin, "location")
+        failover_location = getattrib_wrapped_in_list(plugin, "failoverlocation")
+
+        additional_location_node = find(plugin, "additionalLocations")
+        nodes_list = findall(additional_location_node, "additionalLocation") or []
+        additional_locations = [gettext(node) for node in nodes_list]
+
+        locations = location + additional_locations + failover_location
+        for uri in locations:
```

Comment:
let's set a limit on this... (this is wrt the internal conversation we've been having on network failures and delays)

Reply:
Limit is added, in common/protocol/wire.py

Reply:
thanks